### PR TITLE
Fix the first block computation in export_epoch_info

### PR DESCRIPTION
**Pull request by vkomenda**

### CODE REVIEW

File: src/rpc_extra.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    client: &RpcClient,
+    epoch: Epoch,
+    epoch_info: &EpochInfo,
+    f: F,
+) -> anyhow::Result<Option<A>>
+where
+    F: Fn(u64) -> anyhow::Result<Option<A>>,
+{
+    // First slot in `epoch`.
+    let first_slot = epoch_info.absolute_slot - epoch_info.slot_index;
```
Comment (the-emerald) [inline src/rpc_extra.rs]: Does this not always return the first slot of the current epoch, regardless of the specified epoch number?
 
```suggestion
    let first_slot = epoch * epoch_info.slots_in_epoch;
```

Comment (vkomenda) [reply src/rpc_extra.rs]: Yes, you are right. I was fixing a bug and introduced a regression.

Comment (vkomenda) [reply src/rpc_extra.rs]: And your solution leads to that bug.

File: src/rpc_extra.rs [hunk 2]
```diff
@@ -18,22 +18,12 @@ where
...
-    };
-
     // First block in `epoch`.
-    let first_block = client.get_blocks(first_slot, end_slot)?.get(0).cloned();
+    let first_block = client.get_blocks_with_limit(first_slot, 1)?.get(0).cloned();
 
     if let Some(block) = first_block {
         f(block)
-    } else if end_slot.is_none() {
+    } else if epoch_info.slot_index < SLOT_OFFSET {
```
Comment (the-emerald) [inline src/rpc_extra.rs]: From what I understand, if the slot index is less than the offset then we don't return an error, in the case that there are no blocks?

Comment (vkomenda) [reply src/rpc_extra.rs]: Yes, this is to simulate the previous behaviour where we returned an error only when there were no blocks after `SLOT_OFFSET` was reached.

Comment (the-emerald) [reply src/rpc_extra.rs]: As far as I can tell the slot_offset error is always bubbled up the user, so we won't break anything if we change this. Is there a better API design, perhaps one that doesn't use the `SLOT_OFFSET` at all?

Comment (vkomenda) [reply src/rpc_extra.rs]: We can never return a "no blocks found" error for example.

Comment (the-emerald) [reply src/rpc_extra.rs]: Yes, I think the approach in a14883d is quite nice. Let the error be reserved for more serious errors, and treat the lack of a first block as a `None`.

**Review by the-emerald — APPROVED** [review]

**Pull request merged**